lib/CFRPageActions.jsm

```diff
@@ -118,10 +120,21 @@ class PageAction {
 
   /**
    * Creates a new DOMLocalization instance with the Fluent file from Remote Settings.
-   * Note that it still uses the packaged Fluent file as the fallback if the remote one
-   * is not available.
+   *
+   * Note: it will use the local Fluent file in any of following cases:
+   *   * the remote Fluent file is not available
+   *   * it was told to use the local Fluent file
    */
   _createDOML10n() {
+    if (!Services.prefs.getBoolPref(USE_REMOTE_L10N_PREF, true)) {
+      return new DOMLocalization([
```

Comment:
Oh, I just realized, this probably could have been simplified to:
```js
_createDOML10n() {
  async function* generateBundles() { … }

  return new DOMLocalization(["ftl stuff"], getBoolPref() ? generateBundles : null);
```

Reply:
Doh, I've pushed it to autoland. Will fix it in bug 1588199.